atomdb/species.py

```diff
@@ -62,38 +79,96 @@ def default_matrix():
     return np.zeros(0).reshape(1, 0)
 
 
+# def scalar(method):
+#     r"""Expose a SpeciesData field."""
+#     name = method.__name__
+#
+#     @property
+#     def wrapper(self):
+#         print("hi")
+#
+#         # Map the name of the method in the SpeciesData class to the name in the Elements class
+#         # This dict can be removed if the Elements csv file uses the same names as the SpeciesData class.
+#         namemap = {
+#             "cov_radius": "cov_radius",
+#             "vdw_radius": "vdw_radius",
+#             "at_radius": "at_radius",
+#             "polarizability": "pold",
+#             "dispersion_c6": "c6",
+#             "atmass": "mass",
+#         }
+#
+#         if name == "atmass":
+#             print(f"inside atmass {getattr(Element(self._data.elem), namemap[name])}")
+#             return getattr(Element(self._data.elem), namemap[name])
+#         if name in namemap:
+#             # Only return Element property if neutral, otherwise None
+#             charge = self._data.atnum - self._data.nelec
+#             print(f"charge {charge}")
+#             print(f"inside the other {getattr(Element(self._data.elem), namemap[name])}")
+#             return getattr(Element(self._data.elem), namemap[name]) if charge == 0 else None
+#
+#         return getattr(self._data, name)
+#
+#     # conserve the docstring of the method
+#     wrapper.__doc__ = method.__doc__
+#     return wrapper
+
+
 def scalar(method):
     r"""Expose a SpeciesData field."""
     name = method.__name__
 
     @property
     def wrapper(self):
+        # Checking if the property is not in PROPERTY_NAME_MAP, if not then fetch it from SpeciesData
+        if name not in PROPERTY_NAME_MAP:
+            return getattr(self._data, name)
+
+        # calculate charge then if charge is not zero (ions) --> return none
+        charge = self._data.atnum - self._data.nelec
+        if charge != 0:
+            return None
+
+        # open the HDF5 file in read mode
+        with pt.open_file(elements_hdf5_file, mode="r") as h5file:
+            # get the element group
+            element_group = f"/Elements/{self._data.atnum:03d}"
+
+            table_name = PROPERTY_NAME_MAP[name]
+            table_path = f"{element_group}/{table_name}"
+
+            # get the table node from the HDF5 file
+            table = h5file.get_node(table_path)
+            # get the data type of the 'value' col in the table
+            value_col = table.coltypes.get("value")
+
+            # Handle basic properties (single row)
+            if table.nrows == 1:
+                value = table[0]["value"]
+                # if the value is an int, return it as an int
+                if isinstance(value_col, pt.Int32Col):
```

Comment:
@enjyashraf18 check the type of the `value` variable instead of the `value_col` one, and try using Integral data type from python numbers library. for example like is done here
https://github.com/theochem/AtomDB/blob/464f116f030da1ad05c28ae028e25985ecbed2ea/atomdb/species.py#L417